abdm/tasks/retry_failed_care_contexts.py

```diff
@@ -0,0 +1,38 @@
+from celery import shared_task
+from django.db.models import Q
+
+from abdm.models.abha_number import AbhaNumber
+from abdm.models.transaction import Transaction, TransactionStatus, TransactionType
+from abdm.service.v3.gateway import GatewayService
+
+
+@shared_task
+def retry_failed_care_contexts():
+    unsuccessfull_care_context_transactions = Transaction.objects.filter(
+        ~Q(status=TransactionStatus.COMPLETED),
+        type=TransactionType.LINK_CARE_CONTEXT,
+    )
+
+    for transaction in unsuccessfull_care_context_transactions:
+        if transaction.meta_data.get("type") != "hip_initiated_linking":
+            continue
+
+        abha_id = transaction.meta_data.get("abha_number")
+        abha_number = AbhaNumber.objects.filter(
+            Q(abha_number=abha_id) | Q(health_id=abha_id) | Q(external_id=abha_id)
+        ).first()
+        if not abha_number:
+            continue
+
+        patient = abha_number.patient
+        if not patient:
+            continue
+
+        GatewayService.link__carecontext(
```

Comment:
if we get a rate limit or connection error we should break the loop,

and we could reschedule the task for after the rate limit reset time + some random backoff, waiting for the rate limit for a day would be too much

connection errors could be retried a few times with some exponential backoff

Reply:
added try catch to handle this gracefully, will handle each exception in v3